### ISSUE TITLE
Quantile Regression and Support for Prediction Intervals

### ISSUE THREAD

**Issue by nnorton24** [dmlc/xgboost#1433]
I know that sklearn.ensemble.GradientBoostingRegressor supports quantile regression and the production of [prediction intervals](http://scikit-learn.org/stable/auto_examples/ensemble/plot_gradient_boosting_quantile.html). Are there any plans for the XGBoost package to offer similar support?

**Comment by hetong007** [comment]
Quantile regression is currently not supported. 

It can be achieved by defining the objective function on user's side:  https://github.com/dmlc/xgboost/blob/master/R-package/demo/custom_objective.R

**Comment by arbackus** [comment]
Quantile regression with xgboost requires custom gradient and hessian functions. Here is an implementation in Python: http://www.bigdatarepublic.nl/regression-prediction-intervals-with-xgboost/

**Comment by joseortiz3** [comment]
Note [that](http://www.bigdatarepublic.nl/regression-prediction-intervals-with-xgboost/) implementation is not very useful for most users, since it specifies grid-searching three parameters (very costly) to get a quantile estimate. In sklearn, you define only the quantile value and you are given a very robust and reliable estimate of that quantile.

**Comment by Sandy4321** [comment]
any progress ?

link above is broken 
http://www.bigdatarepublic.nl/regression-prediction-intervals-with-xgboost/

**Comment by trivialfis** [comment]
Related https://github.com/unit8co/darts/issues/1134 .

**Comment by BraulioPi** [comment]
??

**Comment by trivialfis** [comment]
Suppported by the Python package, and R package (still in master).
